lib/sen_tweet/metrics.ex

```diff
@@ -17,7 +17,15 @@ defmodule SenTweet.Metrics do
 
   @impl true
   def init(metrics) do
-    {:ok, metrics}
+    {:ok, tab} = :dets.open_file(:bitcoin_metrics, [type: :set])
+
+    metrics =
+      case :dets.lookup(tab, :metrics) do
```

Comment:
For my understanding...

`:dets` looks like the atom syntax, but it has associated methods? What's going on here?

Reply:
This question is peaking under the veil a bit but nothing special is happening here. It's just modules and functions like normal but `:dets` is an Erlang module. Erlang code is freely available in Elixir and it works because all the code compiles to BEAM bytecode for the VM to execute.

You triggered me by using the word _methods_ so I feel compelled to explain what's happening with these modules and functions. Modules exist to organize, give meaning, and namespaces functions. Functions contain logic and often have arguments that we apply to them. 

```
👉  numbers = [0, 1, 2]
[0, 1, 2]
👉  increment = fn x -> x + 1 end
#Function<7.126501267/1 in :erl_eval.expr/5>
# elixir modules are atoms
👉  is_atom Enum
true
# this is how we typically invoke a function in a module and apply its arguments
👉  Enum.map(numbers, increment)
[1, 2, 3]
# this is what we're really doing above 🤯 
👉  Kernel.apply(Enum, :map, [numbers, increment])
[1, 2, 3]
# now we map using an erlang module
👉  Kernel.apply(:lists, :map, [increment, numbers])
[1, 2, 3]
```